components/kyma-environment-broker/internal/process/provisioning/cls_binding_step.go

```diff
@@ -0,0 +1,171 @@
+package provisioning
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/cls"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/runtime/components"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/servicemanager"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockery --name=ClsBindingProvider --output=automock --outpkg=automock --case=underscore
+type ClsBindingProvider interface {
+	CreateBinding(smClient servicemanager.Client, request *cls.BindingRequest) (*cls.ClsOverrideParams, error)
+}
+
+type ClsBindStep struct {
+	config           *cls.Config
+	operationManager *process.ProvisionOperationManager
+	secretKey        string
+	bindingProvider  ClsBindingProvider
+}
+
+func NewClsBindStep(config *cls.Config, bp ClsBindingProvider, os storage.Operations, secretKey string) *ClsBindStep {
+	return &ClsBindStep{
+		config:           config,
+		operationManager: process.NewProvisionOperationManager(os),
+		secretKey:        secretKey,
+		bindingProvider:  bp,
+	}
+}
+
+var _ Step = (*ClsBindStep)(nil)
+
+func (s *ClsBindStep) Name() string {
+	return "CLS_Bind"
+}
+
+func (s *ClsBindStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if !operation.Cls.Instance.ProvisioningTriggered {
+		failureReason := fmt.Sprintf("cls provisioning step was not triggered")
+		log.Error(failureReason)
+		return s.operationManager.OperationFailed(operation, failureReason)
+	}
+
+	smCredentials, err := cls.FindCredentials(s.config.ServiceManager, operation.Cls.Region)
+	if err != nil {
+		failureReason := fmt.Sprintf("Unable to find credentials for cls service manager in region %s: %s", operation.Cls.Region, err)
+		log.Error(failureReason)
+		return s.operationManager.OperationFailed(operation, failureReason)
+	}
+	smCli := operation.SMClientFactory.ForCredentials(smCredentials)
+
+	// test if the provisioning is finished, if not, retry after 10s
+	resp, err := smCli.LastInstanceOperation(operation.Cls.Instance.InstanceKey(), "")
```

Comment:
if the operation bind succeeded before we don't need to check lastinstanceoperation

Reply:
fixed.